src/array_api_stubs/_draft/array_object.py

```diff
@@ -627,6 +629,10 @@ def __getitem__(
         -------
         out: array
             an array containing the accessed value(s). The returned array must have the same data type as ``self``.
+
+        .. note::
+           When ``__getitem__`` is defined on an object, Python will automatically define iteration (i.e., the behavior from ``iter(x)``) as  ``x[0]``, ``x[1]``, ..., ``x[N-1]``. This can also be implemented directly by defining ``__iter__``. Therefore, for a one-dimensional array ``x``, iteration should produce a sequence of zero-dimensional arrays ``x[0]``, ``x[1]``, ..., ``x[N-1]``, where ``N`` is the number of elements in the array. Iteration behavior for arrays having more than one dimension is unspecified and thus implementation-defined.
```

Comment:
@asmeurer What should happen for zero-dimensional arrays?

Reply:
On a 0-D array `x[i]` is an IndexError, so this doesn't really apply (the Python auto-`__iter__` would never work). Ideally calling `list()` on a 0-D should always be an error, but we can also just change this to say "arrays not having one dimension".